### PR TITLE
Updated Vendored cluster-api and Provider Configurations

### PR DESCRIPTION
**Pull request by marwanad**
The changes introduced here bring the project up-to-date with the upstream cluster-api changes.

The major change introduced is the removal of Machine roles from the Machine Spec. Those can now be specified as part of `AzureMachineProviderConfig`. The config encoders and decoders were also updated to reflect those changes.

### CODE REVIEW

File: cloud/azure/providerconfig/v1alpha1/register.go [hunk 1]
```diff
@@ -24,6 +28,13 @@ const GroupName = "azureproviderconfig"
 
 var SchemeGroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1alpha1"}
 
+// AzureProviderConfigCodec is codec for decoding and encoding DigitalOcean ProviderConfig.
```
Comment (vannrt) [inline cloud/azure/providerconfig/v1alpha1/register.go]: Do you mean Azure ProviderConfig?

File: cloud/azure/providerconfig/types.go [hunk 2]
```diff
@@ -23,13 +23,20 @@ import (
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 type AzureMachineProviderConfig struct {
 	metav1.TypeMeta `json:",inline"`
-	Location        string `json:"location"`
-	VMSize          string `json:"vmSize"`
-	Image           Image  `json:"image"`
-	OSDisk          OSDisk `json:"osDisk"`
-	SSHPublicKey    string `json:"sshPublicKey"`
-	SSHPrivateKey   string `json:"sshPrivateKey"`
+	Roles           []MachineRole `json:"MachineRoles,omitempty"`
```
[the remark this answers is not in the document]
Comment (marwanad) [reply cloud/azure/providerconfig/types.go]: good catch, I missed that

File: Gopkg.toml [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+[prune]
+  go-tests = true
+  unused-packages = true
+
+  [[prune.project]]
+    name = "sigs.k8s.io/cluster-api"
+    unused-packages = false
+
+[[constraint]]
+  branch = "7db708b5df567c79e2e46f9c37cdbf001bef1302"
```
[the remark this answers is not in the document]
Comment (marwanad) [reply Gopkg.toml]: I think I miscommited my vendored directories while testing some changes. 

I re-ran `dep ensure` and committed the proper directories. It should be fixed with 5e003a9.

[the remark this answers is not in the document]
Comment (vannrt) [reply Gopkg.toml]: Thanks @marwanad. Looks good to me. I'm going to run the tests and I'll merge your PR once I get a green build.

**Pull request merged**